File: src/python/src/poly_bus_error.py
"""
Base exception class for PolyBus errors.
"""


class PolyBusError(Exception):
    """Base exception for PolyBus errors."""
    
    def __init__(self, error_code: int, message: str):
        super().__init__(message)
        self.error_code = error_code

File: src/typescript/src/poly-bus-error.ts
/**
 * Base error class for PolyBus errors.
 */
export class PolyBusError extends Error {
  private readonly _errorCode: number;

  /**
   * Creates a new PolyBusError instance.
   * @param errorCode The error code associated with this error.
   * @param message The error message.
   */
  constructor(errorCode: number, message: string) {
    super(message);
    this._errorCode = errorCode;
    this.name = 'PolyBusError';

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, PolyBusError);
    }
  }

  /**
   * Gets the error code associated with this error.
   */
  public get errorCode(): number {
    return this._errorCode;
  }
}
